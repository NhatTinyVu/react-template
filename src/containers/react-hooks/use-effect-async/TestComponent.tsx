import React, { useState } from 'react';
import { useSetStateAfterWaiting } from './TestComponent.hooks';

const period = 2000;

const TestComponent: React.FC = () => {
  const [state, setState] = useState<string>('state is not set');
  useSetStateAfterWaiting({ period, setState });

  return <p>{state}</p>;
};

export default TestComponent;
