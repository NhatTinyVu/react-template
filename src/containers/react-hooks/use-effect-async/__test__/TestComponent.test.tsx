import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TestComponent from '../TestComponent';

describe('EventListener', () => {
  test('renders correctly', (): void => {
    const wrapper = shallow(<TestComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
