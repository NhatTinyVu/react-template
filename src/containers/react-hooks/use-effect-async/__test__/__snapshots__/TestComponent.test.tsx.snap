// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListener renders correctly 1`] = `
<p>
  state is not set
</p>
`;
